src/messages/MOSDPing.h

```diff
@@ -96,11 +100,20 @@ class MOSDPing : public Message {
   void encode_payload(uint64_t features) override {
     ::encode(fsid, payload);
     ::encode(map_epoch, payload);
-    ::encode(peer_as_of_epoch, payload);
-    ::encode(op, payload);
-    ::encode(peer_stat, payload);
+    
+    // with luminous, we drop peer_as_of_epoch and peer_stat
+    if (!HAVE_FEATURE(features, SERVER_LUMINOUS)) { 
+      epoch_t dummy_epoch = {};
+      osd_peer_stat_t dummy_stat = {};
```

Comment:
Is there a reason you're doing the funny initializer?

Yeah, "funny initializer" is how much of some of the newer stuff I really grok. :(